packages/easy-email-extensions/src/AttributePanel/components/blocks/AdvancedTable/Operation/tableTool.ts

```diff
@@ -0,0 +1,244 @@
+    this.root?.addEventListener('contextmenu', this.handleContextmenu);
+    this.root?.addEventListener('mousedown', this.handleMousedown.bind(this));
+    document.body.addEventListener('click', this.hideBorder, false);
+    document.addEventListener('keydown', this.hideBorderByKeyDown);
+  }
+
+  destroy() {
+    this.root?.removeEventListener('mousedown', this.handleMousedown.bind(this));
+    this.root?.removeEventListener('contextmenu', this.handleContextmenu);
+    document.body.removeEventListener('click', this.hideBorder, false);
+    document.removeEventListener('keydown', this.hideBorderByKeyDown);
+  }
+
+  hideBorder = (e: any) => {
+    if (e.target.id === 'VisualEditorEditMode') {
+      return;
+    }
+    this.visibleBorder(false);
+  };
+
+  hideBorderByKeyDown = () => {
+    this.visibleBorder(false);
+  };
+
+  visibleBorder = (show = true) => {
+    if (this.showBorderTool === show) {
+      return;
+    }
+    if (show) {
+      setStyle(this.borderTool.top.parentElement, { display: 'block' });
+    } else {
+      setStyle(this.borderTool.top.parentElement, { display: 'none' });
+    }
+    this.showBorderTool = show;
+  };
+
+  renderBorder = () => {
+    this.visibleBorder(true);
+    const result = getBoundaryRectAndElement(
+      this.startDom as Element,
+      this.endDom as Element,
+    );
+    if (!result) {
+      return;
+    }
+    const { left, top, width, height } = result.boundary;
+    this.selectedLeftTopCell = result.leftTopCell;
+    this.selectedBottomRightCell = result.bottomRightCell;
+
+    setStyle(this.borderTool.top, {
+      'background-color': 'rgb(65, 68, 77)',
+      left: `${left}px`,
+      top: `${top}px`,
+      width: `${Math.abs(width)}px`,
+      height: '2px',
+      position: 'absolute',
+      'z-index': 10,
+    });
+    setStyle(this.borderTool.bottom, {
+      'background-color': 'rgb(65, 68, 77)',
+      left: `${left}px`,
+      top: `${top + height}px`,
+      width: `${Math.abs(width)}px`,
+      height: '2px',
+      position: 'absolute',
+      'z-index': 10,
+    });
+    setStyle(this.borderTool.left, {
+      'background-color': 'rgb(65, 68, 77)',
+      left: `${left}px`,
+      top: `${top}px`,
+      width: `2px`,
+      height: `${Math.abs(height)}px`,
+      position: 'absolute',
+      'z-index': 10,
+    });
+    setStyle(this.borderTool.right, {
+      'background-color': 'rgb(65, 68, 77)',
+      left: `${left + width}px`,
+      top: `${top}px`,
+      width: `2px`,
+      height: `${Math.abs(height)}px`,
+      position: 'absolute',
+      'z-index': 10,
+    });
+  };
+
+  handleContextmenu = (event: any) => {
+    if (this.showBorderTool) {
+      const selectedBoundary = getElementsBoundary(
+        this.selectedLeftTopCell as Element,
+        this.selectedBottomRightCell as Element,
+      );
+      if (checkEventInBoundingRect(selectedBoundary, event)) {
+        event.preventDefault();
+      }
+    }
+  };
```

Comment:
There is a bug here because the default context menu is prevented only when showBorderTool is true. Sometimes, both the custom context menu and the default context menu can appear simultaneously.

Can it be changed to directly display the custom menu, with the currently hovered **td** as the selected location?

Reply:
we cannot display the custom context menu by hover **td**,  because we should  merge selected cells.
I update the code, will hide custom context menu when display default context menu.